src/hotspot/share/gc/serial/defNewGeneration.cpp

```diff
@@ -63,43 +63,75 @@
 #include "utilities/globalDefinitions.hpp"
 #include "utilities/stack.inline.hpp"
 
-//
-// DefNewGeneration functions.
+class IsAliveClosure: public BoolObjectClosure {
+  Generation* _young_gen;
+public:
+  IsAliveClosure(Generation* young_gen) : _young_gen(young_gen) {
+    assert(_young_gen->kind() == Generation::DefNew,
+        "Expected the young generation here");
+  }
 
-// Methods of protected closure types.
+  bool do_object_b(oop p) {
+    return cast_from_oop<HeapWord*>(p) >= _young_gen->reserved().end() || p->is_forwarded();
```

Comment:
Another micro-optimization for another day, `_young_gen->reserved().end()` is constant across the use (i.e. avoid the dereference for `_young_gen->reserved()` all the time).